dashboard/app/models/concerns/partial_registration.rb

```diff
@@ -24,6 +24,14 @@ def new_from_partial_registration(session, &block)
     end
   end
 
+  def self.can_finish_signup?(params, session)
+    if DCDO.get('student-email-post-enabled', false)
+      params&.dig(:user)&.dig(:email)&.present? && in_progress?(session)
```

Comment:
Is it safe to do `params&.dig(:user, :email)`, or is there a possibility that params[:user] doesn't have a `dig` method?

Reply:
Then is it possible to clean this line up by just calling `.dig(:user, :email)` and save the additional `.dig`?